src/main/java/lsd/format/PrettyPrinter.kt

```diff
@@ -0,0 +1,35 @@
+package lsd.format
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import lsd.format.config.log
+import lsd.format.json.indentJson
+import lsd.format.json.objectMapper
+import lsd.format.xml.indentXml
+
+fun prettyPrint(document: Any?): String {
+    if (document == null) return ""
+    if (document is ByteArray) {
+        return indent(String((document as ByteArray?)!!))
+    }
+    return if (document !is String) {
+        objectMapper.writeValueAsString(document)
+    } else indent(document)
+}
+
+private fun indent(doc: String): String = indentJson(doc) ?: indentXml(doc) ?: doc
+
+fun prettyPrintJson(obj: Any?): String? =
+    if (obj == null) {
+        ""
+    } else if (obj is ByteArray && obj.size == 0) {
+        ""
+    } else try {
+        prettyPrint(getDocument(obj))
+    } catch (e: Exception) {
+        log().error("Problem serialising intercepted object for LSD: {}", e.message)
```

Comment:
mixture of log level again - but not sure if this one is more crucial than the others?

Reply:
I think so since it's on the API level.